os/board/common/partitions.c

```diff
@@ -317,9 +324,29 @@ void automount_fs_partition(partition_info_t *partinfo)
 		ret = mount(fs_devname, "/mnt", "smartfs", 0, NULL);
 		if (ret != OK) {
 			lldbg("ERROR: mounting '%s' failed, errno %d\n", fs_devname, get_errno());
+
+#ifdef CONFIG_BOARDCTL_RESET
+			/* Mount was failed, so we'll try low level format forcely */
+			lldbg("Low level format will be done, then board will reset\n");
+
+#ifdef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+			ret = mksmartfs(fs_devname, 1, true);
+#else
+			ret = mksmartfs(fs_devname, true);
+#endif
+			if (ret != OK) {
+				lldbg("ERROR: mksmartfs on %s failed again\n", fs_devname);
```

Comment:
Would it be right to say: -

> mksmartfs failed AGAIN

Since the reason for calling mksmartfs again is failure of `mount `function and not failure of `mksmartfs`

How about the statement: -

> redo/retry mksmartfs failed after mount failure

Reply:
current statement looks fine, as ret is return of mksmartfs not mount, mount ret value already printed